AmplifyPlugins/API/AWSAPICategoryPlugin/Core/AppSyncListProvider.swift

```diff
@@ -7,27 +7,223 @@
+        let predicate: QueryPredicate = field(associatedField) == associatedId
+        let filter = predicate.graphQLFilter(for: Element.schema)
+        let request = GraphQLRequest<JSONValue>.listQuery(responseType: JSONValue.self,
+                                                          modelSchema: Element.schema,
+                                                          filter: filter,
+                                                          limit: limit,
+                                                          apiName: apiName)
+        Amplify.API.query(request: request) { result in
+            switch result {
+            case .success(let graphQLResponse):
+                switch graphQLResponse {
+                case .success(let graphQLData):
+                    guard let listResponse = try? AppSyncListResponse.initWithMetadata(type: Element.self,
+                                                                                       graphQLData: graphQLData,
+                                                                                       apiName: self.apiName) else {
+                        completion(.failure(CoreError.listOperation("""
+                                                The AppSync response return successfully, but could not decode to
+                                                AWSAppSyncListResponse from: \(graphQLData)
+                                                """, "", nil)))
+                        return
+                    }
+
+                    self.loadedState = .loaded(elements: listResponse.items,
+                                               nextToken: listResponse.nextToken,
+                                               filter: filter)
+                    completion(.success(listResponse.items))
+                case .failure(let graphQLError):
+                    Amplify.API.log.error(error: graphQLError)
+                    completion(.failure(.listOperation(
+                                            "The AppSync response returned successfully with GraphQL errors.",
+                                            "Check the underlying error for the failed GraphQL response.",
+                                            graphQLError)))
+                }
+            case .failure(let apiError):
+                Amplify.API.log.error(error: apiError)
+                completion(.failure(.listOperation("The AppSync request failed",
+                                                   "See underlying `APIError` for more details.",
+                                                   apiError)))
+            }
+        }
+    }
+
+    /// Check if there is subsequent data to retrieve. This method always returns false if the underlying provider is
+    /// not loaded. Make sure the underlying data is loaded by calling `load(completion)` before calling this method.
+    /// If true, the next page can be retrieved using `getNextPage(completion:)`.
```

Comment:
nit: Not going to swear to it, but I think Xcode will auto-populate the doc comments from the protocol if you leave the override comments blank. Not a big deal, but it saves you from having to try to keep comments in sync across the different contexts (protocol, different implementations)

Reply:
removing the doc comment optimiscally and seeing, can always add it back if it doesn't, but it sounds like the expected behavior of Xcode